experimental/regression_suite/shark-test-suite-models/llama/test_llama.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import pytest
+from ireers_tools import *
+import os
+from conftest import VmfbManager
+from pathlib import Path
+
+rocm_chip = os.getenv("ROCM_CHIP", default="gfx90a")
+vmfb_dir = os.getenv("TEST_OUTPUT_ARTIFACTS", default=Path.cwd())
+
+###############################################################################
+# Fixtures
+###############################################################################
+llama_mlir = fetch_source_fixture(
+    "https://sharkpublic.blob.core.windows.net/sharkpublic/halo-models/llm-dev/llama3_8b/8b_f16_decomposed_11_22.mlir",
+    group="llama_fp16_8b",
+)
+
+
+ROCM_COMPILE_FLAGS = [
+    "--iree-hal-target-backends=rocm",
+    f"--iree-hip-target={rocm_chip}",
+    "--iree-dispatch-creation-enable-aggressive-fusion=true",
+    "--iree-global-opt-propagate-transposes=true",
+    "--iree-opt-aggressively-propagate-transposes=true",
+    "--iree-opt-data-tiling=false",
+    "--iree-preprocessing-pass-pipeline='builtin.module(util.func(iree-preprocessing-generalize-linalg-matmul-experimental))'",
+    "--iree-stream-resource-memory-model=discrete",
+    "--iree-hip-legacy-sync=false",
+    "--iree-hal-indirect-command-buffers=true",
+    "--iree-hal-memoization=true",
+    "--iree-opt-strip-assertions",
+]
+
+
+###############################################################################
+# CPU
+###############################################################################
+
+# This allows using as part of a matrix strat that includes cpu
+def test_pass_cpu():
```

Comment:
@ScottTodd any thoughts on this? It gives me mildly gross vibes but enables us to add a test here later if desired and makes it so we dont have to branch on the matrix strategy for this suite of tests. Open to alternatives

Reply:
Given the existing test setup, I'd rather use `@pytest.mark.skip("reason") instead of "passing" as a no-op: https://docs.pytest.org/en/stable/how-to/skipping.html

So
```python
@pytest.mark.skip("llama tests not implemented for CPU yet")
def test_compile_llama_cpu():
  pass
```

A better method is used in these older (stale, currently not run) tests, which add marks for the platform and then run using filters: https://github.com/iree-org/iree/blob/5708d428b029504498cb00e19710d7ecc4a89670/experimental/regression_suite/tests/pregenerated/test_llama2.py#L122-L125 https://github.com/iree-org/iree/blob/5708d428b029504498cb00e19710d7ecc4a89670/experimental/regression_suite/tests/pregenerated/test_llama2.py#L145-L148 https://github.com/iree-org/iree/blob/5708d428b029504498cb00e19710d7ecc4a89670/.github/workflows/pkgci_regression_test.yml#L174-L181

That lets new platforms be added in a structured way, without just relying on test case names following a convention.